automol/vmat.py

```diff
@@ -476,24 +451,22 @@ def dihedral_angle_coordinate_name(zma, key1, key2, key3, key4):
 
 
 # # dummy atom functions
-def dummy_keys(zma):
+def dummy_keys(zma: VMatrix) -> tuple[int]:
     """Obtain keys to dummy atoms in the Z-Matrix.
 
     :param zma: Z-Matrix
-    :type zma: automol Z-Matrix data structure
-    :rtype: tuple[int]
+    :return: Key to dummy atoms
     """
     keys = tuple(key for key, sym in enumerate(symbols(zma)) if sym == "X")
     return keys
 
 
-def dummy_coordinate_names(vma):
+def dummy_coordinate_names(vma: VMatrix) -> tuple[str]:
     """Obtain names of all coordinates associated with dummy atoms
     defined in the V-Matrix.
 
     :param vma: V-Matrix
-    :type vma: automol V-Matrix data structure
-    :rtype: tuple(str)
+    :return: Name of coordinates
     """
     symbs = symbols(vma)
     name_mat = numpy.array(name_matrix(vma))
```

Comment:
Looks like you are still working your way through, so I'll stop here. A general thing to be aware of is that the syntax for tuple types is a little weird: unlike for lists, `tuple[str]` refers to a tuple containing only one element, whereas `tuple[str, ...]` refers to a tuple containing any number of elements, all of which are strings.

Reply:
... so you will pretty much always put `tuple[<element type>, ...]` with the ellipses when a function returns a tuple.